### PR TITLE
clubhouse: Play pulse sound only if highlighted button visible

### PR DESCRIPTION
**Pull request by cfoch**
https://phabricator.endlessm.com/T25236

### CODE REVIEW

File: js/ui/components/clubhouse.js [hunk 1]
```diff
@@ -716,21 +723,48 @@ var ClubhouseOpenButton = new Lang.Class({
...
+
+    _play_stop_highlighted_sound() {
+        const states = [
+            SoundServer.SoundItemStatusEnum.NONE,
+            SoundServer.SoundItemStatusEnum.PENDING,
+            SoundServer.SoundItemStatusEnum.CANCELLING
+        ];
+
+        if (this.visible && this.highlighted) {
+            // If the sound item has already an UUID assigned, then do not
```
Comment (manuq) [inline js/ui/components/clubhouse.js]: "UUID assigned" is an implementation detail of the sound server. There are no UUIDs here. I would keep the comment with the method names used here, or remove it.

File: js/ui/components/clubhouse.js [hunk 2]
```diff
@@ -716,21 +723,48 @@ var ClubhouseOpenButton = new Lang.Class({
...
+        this._highlighted = highlighted;
+        this.notify('highlighted');
+    },
+
+    get highlighted() {
+        return this._highlighted;
+    },
+
+    _play_stop_highlighted_sound() {
+        const states = [
```
Comment (manuq) [inline js/ui/components/clubhouse.js]: These states should be defined outside of the callback.

File: js/ui/components/clubhouse.js [hunk 3]
```diff
@@ -716,21 +723,48 @@ var ClubhouseOpenButton = new Lang.Class({
...
             new SoundServer.SoundItem('clubhouse/entry/pulse-loop');
+
+        this.connect('notify::visible',
+                     this._play_stop_highlighted_sound.bind(this));
+        this.connect('notify::highlighted',
+                     this._play_stop_highlighted_sound.bind(this));
     },
 
-    setHighlighted: function(highlighted) {
+    set highlighted(highlighted) {
```
Comment (manuq) [inline js/ui/components/clubhouse.js]: So you are converting the highlighted setter into a property. But should this be set at all if the button is not visible? A simpler approach would be to skip setting the highlighted if the button is hidden, which will also save playing an animation.

Comment (cfoch) [reply js/ui/components/clubhouse.js]: I will opt for this. I was thinking that after completing the InitialContact puzzle the pulsating button would be shown unhighlighted but this is not the case.

**Comment by cfoch** [comment]
I am opting for an easier solution, but just in case I left the old changes in T25236-visibility-dependent

**Pull request merged**